Sources/CircularProgress/CustomButton.swift

```diff
@@ -0,0 +1,327 @@
+import Cocoa
+
+// TODO(sindresorhus): I plan to extract this into a reusable package when it's more mature.
+
+extension CALayer {
+	// TODO: Find a way to use a strongly-typed KeyPath here.
+	// TODO: Accept NSColor instead of CGColor.
```

Comment:
If you have any ideas how to solve these, I would appreciate the help. No worries if not though.